### PR TITLE
feat: allow installation with `requirements.txt`

### PR DESCRIPTION
**Pull request by Luke-zhang-04**
- merge branch upstream with master 33079ad
- docs: update docstirngs to fit with PEP conventions 42ce232
- feat: add `requirements.txt` install function 538d6f2
- feat: users can install with requirements.txt efa4936
- refactor: format code with psf/black 6dd89a0
- docs: add instructions for requirements.txt installation 4fba1c1
- move release to one file, test with and without requirements 668608b
- fix(install_packages): check for array length 325c486
- update examples 6dab958 
- feat: users can install with requirements.txt efa4936
- refactor: format code with psf/black 6dd89a0
- docs: add instructions for requirements.txt installation 4fba1c1
- move release to one file, test with and without requirements 668608b
- fix(install_packages): check for array length 325c486
- update examples 6dab958 
-  fix(CI): fix workflow syntax error on line 36 c55fd36
- refactor: remove f-strings effadfc 
	- f-strings incompatible with Python versions below 3.6, which is the target audience here
-  add pull_request to workflow checking 9015060

### CODE REVIEW

File: examples/ProjectExample/pipinstaller/install_Functions.py [hunk 1]
```diff
@@ -43,9 +41,7 @@ def checkPackage(pkg):
 #Function to install pip
 #######################################
 def installPip(printOut = False):
-    pipPath = resourcesPath / "get-pip.py"
-    print(pipPath)
-    cmd = f"py -3 {pipPath} --user" if sys.platform == "win32" else f"python3 {pipPath} --user"
```
Comment (kertox662) [inline examples/ProjectExample/pipinstaller/install_Functions.py:48]: Wait, what was wrong here?

Comment (kertox662) [reply examples/ProjectExample/pipinstaller/install_Functions.py:48]: might want to add command though instead. f"{cmd} {pipPath}" if this is still valid

File: pipinstaller/install_Functions.py [hunk 2]
```diff
@@ -15,7 +14,8 @@
 #Function to check pip install
 #######################################
 def checkPip():
-    cmd = "pip"
+    cmd = "py -3 -m pip" if sys.platform == 'win32' else "python3 -m pip" #This command will normally give options for
```
Comment (kertox662) [inline pipinstaller/install_Functions.py]: might want to do specific python command instead.
```python
pycmd = "python3"
if sys.platform == 'win32':
    pycmd = "py -3"
cmd = "{cmd} -m pip"
```

Comment (kertox662) [reply pipinstaller/install_Functions.py]: easier to modify later

File: pipinstaller/install_Functions.py [hunk 3]
```diff
@@ -71,7 +70,7 @@ def installPip(printOut = False):
 #Function to install Package
 #######################################
 def installPackage(pkg, printOut = False):
-    cmd = f"pip install {pkg[1]} --user"
+    cmd = "py -3 -m pip install {} --user".format(pkg[1]) if sys.platform == 'win32' else "python3 -m pip install {} --user".format(pkg[1])
```
Comment (kertox662) [inline pipinstaller/install_Functions.py]: same as above, feel f-string might be cleaner.
```python
cmd = f"{pycmd} -m pip install {pkg[1]} --user"
```
I agree though that it should change

**Review by kertox662 — CHANGES_REQUESTED** [review]
The other changes also fall into the same thing. I do think using the path module is cleaner than hardcoding the path with the separators as strings, don't know if there's any specific issue with it.

File: examples/ProjectExample/pipinstaller/install_Functions.py [hunk 4]
```diff
@@ -25,76 +23,83 @@ def checkPip():
...
     try:
         _ = __import__(pkg[0])
-    except ImportError: #If the PIL package is not installed, this import will raise an ImportError
+    except ImportError: # If the package in question is not installed, this import will raise an ImportError
         packagesMissing.append(pkg)
         return False
-    print(pkg[0],"was found on the system")
-    return True #Returns if whether it is installed or not
+    print(pkg[0], "was found on the system")
+    return True # Returns if whether it is installed or not
```
Comment (kertox662) [inline examples/ProjectExample/pipinstaller/install_Functions.py]: Small thing, could you quickly change it to "Returns whether it ...", just remove the if. I think I had something there before, doesn't make sense anymore.

**Review by kertox662 — APPROVED** [review]

**Comment by kertox662** [comment]
Just curious if you know why all 20 commits are opened up now instead of just the modified ones. Is it because of commit squashing? I'm pretty sure though it should only be 2 or so if I'm not wrong.

**Comment by Luke-zhang-04** [comment]
> Just curious if you know why all 20 commits are opened up now instead of just the modified ones. Is it because of commit squashing? I'm pretty sure though it should only be 2 or so if I'm not wrong.

I don't know

File: pipinstaller/install_Functions.py (hunk 3 above)
Comment (Luke-zhang-04) [reply pipinstaller/install_Functions.py]: F strings are not supported in python 3.5 and below
[https://www.python.org/dev/peps/pep-0498/](https://www.python.org/dev/peps/pep-0498/)
See effadfc

**Pull request merged**